### PR TITLE
Sim manager refactoring

### PR DESCRIPTION
**Pull request by la3lma**
1. Fix the ES2+ ping protocol used to check if the SM-DP+ in the other end is answering requests.
2. One round of refactoring of ProfileVendorAdapter.kt.  This round of refactoring consists of 
    a) Introducing a set of methods that factor out a lot of repeated code.
    b) Removing some functionality (in particular with respect to eid), that is not necessary for our current usecases, and maybe also not correct.  All in all, this simplification makes the code much more compact and easy to read.

### CODE REVIEW

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SmdpPlusHealthceck.kt [hunk 1]
```diff
@@ -82,9 +82,23 @@ class SmdpPlusHealthceck(
                                 httpClient = httpClient,
                                 config = currentConfig!!
```
Comment (kmmatwork) [inline sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SmdpPlusHealthceck.kt]: !! only allowed in tests.... :-)

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SmdpPlusHealthceck.kt [hunk 2]
```diff
@@ -68,7 +68,7 @@ class SmdpPlusHealthceck(
 
                     val profileVendorAdaptorList = vendorsRaw.bind()
```
Comment (kmmatwork) [inline sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SmdpPlusHealthceck.kt:71]: Not visible in the diff, but I don't think that wrapping the `checkIfSmdpPlusIsUp` method in a try .. catch is necessary, as all calls returns an Either?

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SmdpPlusHealthceck.kt (hunk 1 above)
Comment (la3lma) [reply sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SmdpPlusHealthceck.kt]: Fixed in next commit.

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SmdpPlusHealthceck.kt (hunk 2 above)
Comment (la3lma) [reply sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SmdpPlusHealthceck.kt:71]: Maybe.  Will look into it.  There is a _lot_ of refactoring left in this code.

Comment (la3lma) [reply sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SmdpPlusHealthceck.kt:71]: For now adding a TODO, but I will get back to it in a later PR.

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/profilevendors/ProfileVendorAdapter.kt [hunk 3]
```diff
@@ -162,101 +177,112 @@ data class ProfileVendorAdapter(
...
-                        logger.error("SM-DP+ 'order-confirm' message to service {} for ICCID {} failed with status code {} (call-id: {})",
-                                config.name,
-                                simEntry.iccid,
-                                it?.statusLine?.statusCode,
-                                header.functionCallIdentifier)
-                        NotUpdatedError("SM-DP+ 'order-confirm' to ${config.name} failed with code: ${it.statusLine.statusCode}")
-                                .left()
-                    }
+                when {
+                    simEntryId == null -> AdapterError("simEntryId == null").left()
```
Comment (kmmatwork) [inline sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/profilevendors/ProfileVendorAdapter.kt]: Not sure about this, as the `simEntryId` and `statusId` goes out of scope after the `else` part... 

Maybe there should be a `return` here if either or both of the two variables are null?

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/profilevendors/ProfileVendorAdapter.kt [hunk 4]
```diff
@@ -162,101 +177,112 @@ data class ProfileVendorAdapter(
...
+            logger.info("SM-DP+ 'order-confirm' message to service {} for ICCID {} completed OK (call-id: {})",
+                    nameOfSimVendor,
+                    iccid,
+                    functionCallIdentifier)
+
+            val simEntryId = simEntry.id
+            val statusMatchingId = status.matchingId
+
+            when {
+                simEntryId == null -> AdapterError("simEntryId == null").left()
```
Comment (kmmatwork) [inline sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/profilevendors/ProfileVendorAdapter.kt]: A bit "thin" error messages here.

**Review by kmmatwork — APPROVED** [review]
LGTM.

But could overall benefit from a bit of rework to make things clearer (see comments). Plus, have not looked at the Go code.

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/profilevendors/ProfileVendorAdapter.kt (hunk 4 above)
Comment (la3lma) [reply sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/profilevendors/ProfileVendorAdapter.kt]: Tar det som en kommentar inn i neste runde av refactoring, fikser det ikke akkurat nå (legger det som en "TODO" i koden nå, men pusher ikke til denne PR.

File: sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/profilevendors/ProfileVendorAdapter.kt (hunk 3 above)
Comment (la3lma) [reply sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/profilevendors/ProfileVendorAdapter.kt]: Have simplified the code.

**Pull request merged**